external-crates/move/crates/move-compiler/src/linters/mod.rs

```diff
@@ -17,16 +20,33 @@ pub enum LintLevel {
 
 pub const ALLOW_ATTR_CATEGORY: &str = "lint";
 pub const LINT_WARNING_PREFIX: &str = "Lint ";
+pub const REDUNDANT_BOOLEAN_EXP_FILTER_NAME: &str = "redundant_boolean_expressions";
+
+pub const LINTER_DEFAULT_DIAG_CODE: u8 = 1;
+
+pub enum LinterDiagCategory {
+    RedundantBooleanExp,
+}
```

Comment:
See comment on #16488, we should make sure we have a good conceptual split between category and code